artiq/examples/kasli_basic/repository/kasli_tester.py

```diff
@@ -16,11 +18,18 @@ def chunker(seq, size):
 
 
 def is_enter_pressed() -> TBool:
-    if select.select([sys.stdin,], [], [], 0.0)[0]:
-        sys.stdin.read(1)
-        return True
-    else:
-        return False
+    if sys.platform == 'linux2':
+        if select.select([sys.stdin,], [], [], 0.0)[0]:
+            sys.stdin.read(1)
+            return True
+        else:
+            return False        
+    elif sys.platform == 'win32':   
+        if msvcrt.kbhit():
+            sys.stdin.read(1)
```

Comment:
Doesn't this block (and break) if the key pressed is not ENTER?

Reply:
On Linux, the ``select`` call waits for ENTER, and ``sys.stdin.read`` also waits for ENTER.
On Windows, the ``kbhit`` call does **NOT** wait for ENTER, and I suspect ``sys.stdin.read`` may still wait for ENTER.

Reply:
Yes, on windows `kbhit()` will immditely return `0`, which is not `True`. But if you press any button, a none-zero value will return, which is `True`.
Then `sys.stdin.read(1)` will block untill ENTER.

Reply:
Then the behavior is incorrect. You may want to call ``msvcrt.getch`` on Windows instead (and test for the ENTER key so it's consistent with Linux).

Reply:
The problem is: if the user presses a key other than ENTER and nothing else, then the program freezes.
Characters displayed on the console are not important.

Reply:
> On Linux, the `select` call waits for ENTER, and `sys.stdin.read` also waits for ENTER.

Just want to clarify that on linux, did the `sys.stdin.read(1)` get the last line from `select` or it started wait for a new line?
It sounds like I need to press ENTER twice to `return True`

Reply:
``select`` does not consume any characters, it only checks if something can be read.
So, pressing once is sufficient.